### PR TITLE
finalizing draft for riseupspec

### PR DESCRIPTION
**Pull request by cyBerta**

### CODE REVIEW

File: nettests/ts-026-riseupvpn.md [hunk 1]
```diff
@@ -9,47 +9,46 @@ RiseupVPN
...
 
 1. The RiseupVPN API service
-1. The advertised gateways (the VPN servers used by RiseupVPN clients)
+2. The advertised gateways (the VPN servers used by RiseupVPN clients)
+
 
 ## RiseupVPN API check
 
-The RiseupVPN provider serves a JSON describing which servers a client could use (https://example.org/provider.json). When parsing, we learn where their CA-certificate is stored. The locations of the VPN gateway servers are also described. At last we will request a client certificate, enabling us to proceed to the next tests. This all implies a working DNS.
+A HTTP GET request to a commercial CA certified API endpoint will hand out the self-signed certificate needed to access the provider API. While RiseupVPN clients will perform a fingerprint verification, this test skips that step in order to reduce the complexity.
```
Comment (KwadroNaut) [inline nettests/ts-026-riseupvpn.md]: Can't think of a better description as 'commercial CA,' but if you have one (Mozilla? Defaults in ...?), we should change it. If we don't let's just keep this. Or maybe just write 'GET request to an API endpoint which will hand out…' ?

File: nettests/ts-026-riseupvpn.md [hunk 2]
```diff
@@ -9,47 +9,46 @@ RiseupVPN
...
+
+The RiseupVPN provider serves a JSON describing which endpoints a client could use (https://example.org/provider.json) for bootstrapping. However, for the sake of simplicity we hardcoded all endpoints that will be tested. 
+Using the self signed certificate, we perform HTTP GET requests to test the reachability of the provider api endpoints. The API check implies a working DNS.
```
Comment (KwadroNaut) [inline nettests/ts-026-riseupvpn.md]: Let's write API consistently in all capitals?

File: nettests/ts-026-riseupvpn.md [hunk 3]
```diff
@@ -58,29 +57,39 @@ If all parts of the API are functional and reachable then we write:
...
-    "riseupvpn_api_status": "ok"
+    "riseupvpn_api_status": "ok",
+    "riseupvpn_ca_cert_status": true,
 }
 ```
 
 ## RiseupVPN gateways test
 
-When OpenVPN is installed, we can verify if a connection to these gateways is possible. If they can't be reached or we don't get the expected results we can presume them to be blocked. However, without OpenVPN installed, we don't run this test. We test by using the client certificate fetched in the API check.
+If the provider API is reachable, it provides a JSON-file which contains the IP addresses and capabilites of the VPN gateways. The reachability of gateways will be tested depending on their capabilities as described by the provider (ports, OpenVPN, obfs4) by performing TCP handshakes. If a TCP handshake fails we can presume the corresponding port and transport of that gateway to be blocked.
```
Comment (KwadroNaut) [inline nettests/ts-026-riseupvpn.md]: assume, not presume :)

File: nettests/ts-026-riseupvpn.md [hunk 4]
```diff
@@ -58,29 +57,39 @@ If all parts of the API are functional and reachable then we write:
...
 
-If any one of the OpenVPN gateways are blocked then we presume them to blocked and write in the report: FIXME: DO WE WRITE WHICH GATEWAYS ARE BLOCKED? HOW?
+Example output for reported blocked gateways:
 
 ```json
 {
-    "riseupvpn_gateway_failure": "FAILURE STRING",
-    "riseupvpn_gateway_status": "blocked"
+   "riseupvpn_failing_gateways":[
```
Comment (KwadroNaut) [inline nettests/ts-026-riseupvpn.md:73]: Shouldn't we use 192.0.2.0/24  as example IPs instead of real ones? It's nitpicking.

**Review by KwadroNaut — COMMENTED** [review]
It's looking good, only have minor comments.

File: nettests/ts-026-riseupvpn.md (hunk 1 above)
Comment (cyBerta) [reply nettests/ts-026-riseupvpn.md]: > Or maybe just write 'GET request to an API endpoint which will hand out…' ?

sounds good

File: nettests/ts-026-riseupvpn.md (hunk 4 above)
Comment (cyBerta) [reply nettests/ts-026-riseupvpn.md:73]: ok

**Pull request merged**